{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Trying to analyse iPSC data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import csv\n",
    "import numpy as np\n",
    "import scipy\n",
    "import pickle\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import random\n",
    "\n",
    "\n",
    "\n",
    "from numpy import loadtxt\n",
    "from xgboost import XGBClassifier\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import accuracy_score\n",
    "\n",
    "from sklearn import preprocessing\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.metrics import classification_report, confusion_matrix\n",
    "from sklearn.metrics import precision_recall_curve\n",
    "from sklearn.metrics import roc_auc_score\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import average_precision_score\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "from sklearn import datasets\n",
    "from sklearn.metrics import roc_curve, auc\n",
    "from sklearn.multiclass import OneVsRestClassifier\n",
    "from scipy import interp\n",
    "from sklearn import metrics\n",
    "\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.decomposition import PCA\n",
    "\n",
    "import requests\n",
    "import itertools\n",
    "import networkx as nx\n",
    "import datetime\n",
    "import io\n",
    "import random\n",
    "import warnings\n",
    "import json\n",
    "import requests\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "metadata": {},
   "outputs": [],
   "source": [
    "#iPSC data from here \n",
    "#https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE79636\n",
    "\n",
    "#Paper is from here \n",
    "#https://www.sciencedirect.com/science/article/pii/S1934590916304015?via%3Dihub#bib13\n",
    "\n",
    "#transcript length from here\n",
    "# https://www.ensembl.org/biomart/martview/1143c37a535d4b67048047cdd1a16ad4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob\n",
    "files = (glob.glob(\"./iPSC/*_counts.txt\"))\n",
    "ipsc_samples_df = []\n",
    "for i in range(len(files)):\n",
    "    file_url = files[i]\n",
    "    col_name = file_url.split('/')[-1].split('.')[0]\n",
    "    sample_df =  pd.read_csv(file_url,sep='\\t',header = None)\n",
    "    sample_df = sample_df.rename(columns = {0:'Gene',1:col_name})\n",
    "    if i == 0:\n",
    "        ipsc_samples_df = sample_df\n",
    "    else:\n",
    "        ipsc_samples_df = pd.merge(ipsc_samples_df,sample_df, how= 'inner', on = 'Gene')\n",
    "    \n",
    "        \n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(56632, 318)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>GSM2100046_291_1_counts</th>\n",
       "      <th>GSM2099989_107_5_counts</th>\n",
       "      <th>GSM2100039_255_2_counts</th>\n",
       "      <th>GSM2100240_845_6_counts</th>\n",
       "      <th>GSM2100242_864_4_counts</th>\n",
       "      <th>GSM2100193_821_8_counts</th>\n",
       "      <th>GSM2100273_998_4_counts</th>\n",
       "      <th>GSM2099965_024_1_counts</th>\n",
       "      <th>GSM2100069_352_1_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM2100229_839_1_counts</th>\n",
       "      <th>GSM2100064_345_1_counts</th>\n",
       "      <th>GSM2100032_243_1_counts</th>\n",
       "      <th>GSM2099966_024_4_counts</th>\n",
       "      <th>GSM2100230_839_5_counts</th>\n",
       "      <th>GSM2100170_601_4_counts</th>\n",
       "      <th>GSM2100107_799_03_counts</th>\n",
       "      <th>GSM2100182_668_1_counts</th>\n",
       "      <th>GSM2099968_054_1_counts</th>\n",
       "      <th>GSM2100023_215_1_counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 318 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  GSM2100046_291_1_counts  GSM2099989_107_5_counts  \\\n",
       "0  ENSG00000210049                        0                        2   \n",
       "\n",
       "   GSM2100039_255_2_counts  GSM2100240_845_6_counts  GSM2100242_864_4_counts  \\\n",
       "0                        4                        1                        4   \n",
       "\n",
       "   GSM2100193_821_8_counts  GSM2100273_998_4_counts  GSM2099965_024_1_counts  \\\n",
       "0                        7                        2                        2   \n",
       "\n",
       "   GSM2100069_352_1_counts  ...  GSM2100229_839_1_counts  \\\n",
       "0                        2  ...                        2   \n",
       "\n",
       "   GSM2100064_345_1_counts  GSM2100032_243_1_counts  GSM2099966_024_4_counts  \\\n",
       "0                        5                        4                        4   \n",
       "\n",
       "   GSM2100230_839_5_counts  GSM2100170_601_4_counts  GSM2100107_799_03_counts  \\\n",
       "0                        5                        7                         1   \n",
       "\n",
       "   GSM2100182_668_1_counts  GSM2099968_054_1_counts  GSM2100023_215_1_counts  \n",
       "0                        1                        2                        1  \n",
       "\n",
       "[1 rows x 318 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(ipsc_samples_df.shape)\n",
    "ipsc_samples_df.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(66832, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>transcript_length</th>\n",
       "      <th>Gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25874</th>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>3796</td>\n",
       "      <td>TSPAN6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25791</th>\n",
       "      <td>ENSG00000000005</td>\n",
       "      <td>1205</td>\n",
       "      <td>TNMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101154</th>\n",
       "      <td>ENSG00000000419</td>\n",
       "      <td>1161</td>\n",
       "      <td>DPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30085</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>6308</td>\n",
       "      <td>SCYL3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29803</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>4355</td>\n",
       "      <td>C1orf112</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Gene  transcript_length Gene_name\n",
       "25874   ENSG00000000003               3796    TSPAN6\n",
       "25791   ENSG00000000005               1205      TNMD\n",
       "101154  ENSG00000000419               1161      DPM1\n",
       "30085   ENSG00000000457               6308     SCYL3\n",
       "29803   ENSG00000000460               4355  C1orf112"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript_lengths = pd.read_csv('iPSC/transcript_lengths.txt',sep='\\t')\n",
    "transcript_lengths = transcript_lengths.rename(columns = {'Gene stable ID':'Gene', 'Gene name':'Gene_name',\n",
    "                                                          'Transcript length (including UTRs and CDS)':'transcript_length'})\n",
    "\n",
    "transcript_lengths = transcript_lengths.sort_values(['Gene', 'transcript_length'], ascending=[1, 0])\n",
    "transcript_lengths = transcript_lengths.drop_duplicates(subset=['Gene'], keep='first')\n",
    "print(transcript_lengths.shape)\n",
    "transcript_lengths.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(50173, 320)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>GSM2100046_291_1_counts</th>\n",
       "      <th>GSM2099989_107_5_counts</th>\n",
       "      <th>GSM2100039_255_2_counts</th>\n",
       "      <th>GSM2100240_845_6_counts</th>\n",
       "      <th>GSM2100242_864_4_counts</th>\n",
       "      <th>GSM2100193_821_8_counts</th>\n",
       "      <th>GSM2100273_998_4_counts</th>\n",
       "      <th>GSM2099965_024_1_counts</th>\n",
       "      <th>GSM2100069_352_1_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM2100032_243_1_counts</th>\n",
       "      <th>GSM2099966_024_4_counts</th>\n",
       "      <th>GSM2100230_839_5_counts</th>\n",
       "      <th>GSM2100170_601_4_counts</th>\n",
       "      <th>GSM2100107_799_03_counts</th>\n",
       "      <th>GSM2100182_668_1_counts</th>\n",
       "      <th>GSM2099968_054_1_counts</th>\n",
       "      <th>GSM2100023_215_1_counts</th>\n",
       "      <th>transcript_length</th>\n",
       "      <th>Gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>71</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>9613</td>\n",
       "      <td>7375</td>\n",
       "      <td>19474</td>\n",
       "      <td>18168</td>\n",
       "      <td>15023</td>\n",
       "      <td>42973</td>\n",
       "      <td>20188</td>\n",
       "      <td>8444</td>\n",
       "      <td>7442</td>\n",
       "      <td>...</td>\n",
       "      <td>13932</td>\n",
       "      <td>10686</td>\n",
       "      <td>27189</td>\n",
       "      <td>13735</td>\n",
       "      <td>10818</td>\n",
       "      <td>9832</td>\n",
       "      <td>10821</td>\n",
       "      <td>17674</td>\n",
       "      <td>954</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>69</td>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>90984</td>\n",
       "      <td>90983</td>\n",
       "      <td>123844</td>\n",
       "      <td>153154</td>\n",
       "      <td>240328</td>\n",
       "      <td>726493</td>\n",
       "      <td>123442</td>\n",
       "      <td>103665</td>\n",
       "      <td>50372</td>\n",
       "      <td>...</td>\n",
       "      <td>116376</td>\n",
       "      <td>79048</td>\n",
       "      <td>231812</td>\n",
       "      <td>170231</td>\n",
       "      <td>122754</td>\n",
       "      <td>130937</td>\n",
       "      <td>151396</td>\n",
       "      <td>71703</td>\n",
       "      <td>1559</td>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 320 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  GSM2100046_291_1_counts  GSM2099989_107_5_counts  \\\n",
       "0  ENSG00000210049                        0                        2   \n",
       "1  ENSG00000211459                     9613                     7375   \n",
       "2  ENSG00000210077                        0                        0   \n",
       "3  ENSG00000210082                    90984                    90983   \n",
       "4  ENSG00000209082                        0                        0   \n",
       "\n",
       "   GSM2100039_255_2_counts  GSM2100240_845_6_counts  GSM2100242_864_4_counts  \\\n",
       "0                        4                        1                        4   \n",
       "1                    19474                    18168                    15023   \n",
       "2                        0                        0                        0   \n",
       "3                   123844                   153154                   240328   \n",
       "4                        0                        0                        0   \n",
       "\n",
       "   GSM2100193_821_8_counts  GSM2100273_998_4_counts  GSM2099965_024_1_counts  \\\n",
       "0                        7                        2                        2   \n",
       "1                    42973                    20188                     8444   \n",
       "2                        0                        0                        0   \n",
       "3                   726493                   123442                   103665   \n",
       "4                        0                        0                        0   \n",
       "\n",
       "   GSM2100069_352_1_counts  ...  GSM2100032_243_1_counts  \\\n",
       "0                        2  ...                        4   \n",
       "1                     7442  ...                    13932   \n",
       "2                        0  ...                        0   \n",
       "3                    50372  ...                   116376   \n",
       "4                        0  ...                        0   \n",
       "\n",
       "   GSM2099966_024_4_counts  GSM2100230_839_5_counts  GSM2100170_601_4_counts  \\\n",
       "0                        4                        5                        7   \n",
       "1                    10686                    27189                    13735   \n",
       "2                        0                        0                        1   \n",
       "3                    79048                   231812                   170231   \n",
       "4                        0                        0                        0   \n",
       "\n",
       "   GSM2100107_799_03_counts  GSM2100182_668_1_counts  GSM2099968_054_1_counts  \\\n",
       "0                         1                        1                        2   \n",
       "1                     10818                     9832                    10821   \n",
       "2                         0                        0                        0   \n",
       "3                    122754                   130937                   151396   \n",
       "4                         0                        0                        0   \n",
       "\n",
       "   GSM2100023_215_1_counts  transcript_length  Gene_name  \n",
       "0                        1                 71      MT-TF  \n",
       "1                    17674                954    MT-RNR1  \n",
       "2                        0                 69      MT-TV  \n",
       "3                    71703               1559    MT-RNR2  \n",
       "4                        0                 75     MT-TL1  \n",
       "\n",
       "[5 rows x 320 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ipsc_samples_df = pd.merge(ipsc_samples_df, transcript_lengths, how='inner', on='Gene')\n",
    "print(ipsc_samples_df.shape)\n",
    "ipsc_samples_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>GSM2100046_291_1_counts</th>\n",
       "      <th>GSM2099989_107_5_counts</th>\n",
       "      <th>GSM2100039_255_2_counts</th>\n",
       "      <th>GSM2100240_845_6_counts</th>\n",
       "      <th>GSM2100242_864_4_counts</th>\n",
       "      <th>GSM2100193_821_8_counts</th>\n",
       "      <th>GSM2100273_998_4_counts</th>\n",
       "      <th>GSM2099965_024_1_counts</th>\n",
       "      <th>GSM2100069_352_1_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM2100032_243_1_counts</th>\n",
       "      <th>GSM2099966_024_4_counts</th>\n",
       "      <th>GSM2100230_839_5_counts</th>\n",
       "      <th>GSM2100170_601_4_counts</th>\n",
       "      <th>GSM2100107_799_03_counts</th>\n",
       "      <th>GSM2100182_668_1_counts</th>\n",
       "      <th>GSM2099968_054_1_counts</th>\n",
       "      <th>GSM2100023_215_1_counts</th>\n",
       "      <th>transcript_length</th>\n",
       "      <th>Gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>0.056338</td>\n",
       "      <td>0.014085</td>\n",
       "      <td>0.056338</td>\n",
       "      <td>0.098592</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>...</td>\n",
       "      <td>0.056338</td>\n",
       "      <td>0.056338</td>\n",
       "      <td>0.070423</td>\n",
       "      <td>0.098592</td>\n",
       "      <td>0.014085</td>\n",
       "      <td>0.014085</td>\n",
       "      <td>0.028169</td>\n",
       "      <td>0.014085</td>\n",
       "      <td>71</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>10.076520</td>\n",
       "      <td>7.730608</td>\n",
       "      <td>20.412998</td>\n",
       "      <td>19.044025</td>\n",
       "      <td>15.747379</td>\n",
       "      <td>45.045073</td>\n",
       "      <td>21.161426</td>\n",
       "      <td>8.851153</td>\n",
       "      <td>7.800839</td>\n",
       "      <td>...</td>\n",
       "      <td>14.603774</td>\n",
       "      <td>11.201258</td>\n",
       "      <td>28.500000</td>\n",
       "      <td>14.397275</td>\n",
       "      <td>11.339623</td>\n",
       "      <td>10.306080</td>\n",
       "      <td>11.342767</td>\n",
       "      <td>18.526205</td>\n",
       "      <td>954</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.014493</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>69</td>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>58.360487</td>\n",
       "      <td>58.359846</td>\n",
       "      <td>79.438101</td>\n",
       "      <td>98.238614</td>\n",
       "      <td>154.155228</td>\n",
       "      <td>465.999359</td>\n",
       "      <td>79.180244</td>\n",
       "      <td>66.494548</td>\n",
       "      <td>32.310455</td>\n",
       "      <td>...</td>\n",
       "      <td>74.647851</td>\n",
       "      <td>50.704298</td>\n",
       "      <td>148.692752</td>\n",
       "      <td>109.192431</td>\n",
       "      <td>78.738935</td>\n",
       "      <td>83.987813</td>\n",
       "      <td>97.110969</td>\n",
       "      <td>45.992944</td>\n",
       "      <td>1559</td>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>75</td>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 320 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  GSM2100046_291_1_counts  GSM2099989_107_5_counts  \\\n",
       "0  ENSG00000210049                 0.000000                 0.028169   \n",
       "1  ENSG00000211459                10.076520                 7.730608   \n",
       "2  ENSG00000210077                 0.000000                 0.000000   \n",
       "3  ENSG00000210082                58.360487                58.359846   \n",
       "4  ENSG00000209082                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100039_255_2_counts  GSM2100240_845_6_counts  GSM2100242_864_4_counts  \\\n",
       "0                 0.056338                 0.014085                 0.056338   \n",
       "1                20.412998                19.044025                15.747379   \n",
       "2                 0.000000                 0.000000                 0.000000   \n",
       "3                79.438101                98.238614               154.155228   \n",
       "4                 0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100193_821_8_counts  GSM2100273_998_4_counts  GSM2099965_024_1_counts  \\\n",
       "0                 0.098592                 0.028169                 0.028169   \n",
       "1                45.045073                21.161426                 8.851153   \n",
       "2                 0.000000                 0.000000                 0.000000   \n",
       "3               465.999359                79.180244                66.494548   \n",
       "4                 0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100069_352_1_counts  ...  GSM2100032_243_1_counts  \\\n",
       "0                 0.028169  ...                 0.056338   \n",
       "1                 7.800839  ...                14.603774   \n",
       "2                 0.000000  ...                 0.000000   \n",
       "3                32.310455  ...                74.647851   \n",
       "4                 0.000000  ...                 0.000000   \n",
       "\n",
       "   GSM2099966_024_4_counts  GSM2100230_839_5_counts  GSM2100170_601_4_counts  \\\n",
       "0                 0.056338                 0.070423                 0.098592   \n",
       "1                11.201258                28.500000                14.397275   \n",
       "2                 0.000000                 0.000000                 0.014493   \n",
       "3                50.704298               148.692752               109.192431   \n",
       "4                 0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100107_799_03_counts  GSM2100182_668_1_counts  GSM2099968_054_1_counts  \\\n",
       "0                  0.014085                 0.014085                 0.028169   \n",
       "1                 11.339623                10.306080                11.342767   \n",
       "2                  0.000000                 0.000000                 0.000000   \n",
       "3                 78.738935                83.987813                97.110969   \n",
       "4                  0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100023_215_1_counts  transcript_length  Gene_name  \n",
       "0                 0.014085                 71      MT-TF  \n",
       "1                18.526205                954    MT-RNR1  \n",
       "2                 0.000000                 69      MT-TV  \n",
       "3                45.992944               1559    MT-RNR2  \n",
       "4                 0.000000                 75     MT-TL1  \n",
       "\n",
       "[5 rows x 320 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for column in ipsc_samples_df.columns:\n",
    "    if column not in ['Gene','transcript_length' ,'Gene_name']:\n",
    "        ipsc_samples_df[column] =  ipsc_samples_df[column]/ ipsc_samples_df['transcript_length']\n",
    "\n",
    "ipsc_samples_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "scaling_fac_dic = {}\n",
    "\n",
    "for column in ipsc_samples_df.columns:\n",
    "    if column not in ['Gene','transcript_length' ,'Gene_name']:\n",
    "        scaling_fac = sum(ipsc_samples_df[column].tolist())/10**6\n",
    "        scaling_fac_dic[column] = scaling_fac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>GSM2100046_291_1_counts</th>\n",
       "      <th>GSM2099989_107_5_counts</th>\n",
       "      <th>GSM2100039_255_2_counts</th>\n",
       "      <th>GSM2100240_845_6_counts</th>\n",
       "      <th>GSM2100242_864_4_counts</th>\n",
       "      <th>GSM2100193_821_8_counts</th>\n",
       "      <th>GSM2100273_998_4_counts</th>\n",
       "      <th>GSM2099965_024_1_counts</th>\n",
       "      <th>GSM2100069_352_1_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM2100032_243_1_counts</th>\n",
       "      <th>GSM2099966_024_4_counts</th>\n",
       "      <th>GSM2100230_839_5_counts</th>\n",
       "      <th>GSM2100170_601_4_counts</th>\n",
       "      <th>GSM2100107_799_03_counts</th>\n",
       "      <th>GSM2100182_668_1_counts</th>\n",
       "      <th>GSM2099968_054_1_counts</th>\n",
       "      <th>GSM2100023_215_1_counts</th>\n",
       "      <th>transcript_length</th>\n",
       "      <th>Gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000210049</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.769992</td>\n",
       "      <td>6.247799</td>\n",
       "      <td>1.786674</td>\n",
       "      <td>5.713135</td>\n",
       "      <td>5.205140</td>\n",
       "      <td>2.829007</td>\n",
       "      <td>2.867638</td>\n",
       "      <td>2.821041</td>\n",
       "      <td>...</td>\n",
       "      <td>7.064852</td>\n",
       "      <td>7.307869</td>\n",
       "      <td>6.633206</td>\n",
       "      <td>11.450310</td>\n",
       "      <td>1.259363</td>\n",
       "      <td>1.292854</td>\n",
       "      <td>2.895095</td>\n",
       "      <td>1.489433</td>\n",
       "      <td>71</td>\n",
       "      <td>MT-TF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000211459</td>\n",
       "      <td>926.463645</td>\n",
       "      <td>1034.623715</td>\n",
       "      <td>2263.769598</td>\n",
       "      <td>2415.808088</td>\n",
       "      <td>1596.912571</td>\n",
       "      <td>2378.154201</td>\n",
       "      <td>2125.236620</td>\n",
       "      <td>901.057693</td>\n",
       "      <td>781.230165</td>\n",
       "      <td>...</td>\n",
       "      <td>1831.329630</td>\n",
       "      <td>1452.967437</td>\n",
       "      <td>2684.458441</td>\n",
       "      <td>1672.083082</td>\n",
       "      <td>1013.929710</td>\n",
       "      <td>946.021915</td>\n",
       "      <td>1165.762835</td>\n",
       "      <td>1959.141027</td>\n",
       "      <td>954</td>\n",
       "      <td>MT-RNR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000210077</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.683172</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>69</td>\n",
       "      <td>MT-TV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000210082</td>\n",
       "      <td>5365.827728</td>\n",
       "      <td>7810.573375</td>\n",
       "      <td>8809.561419</td>\n",
       "      <td>12461.947384</td>\n",
       "      <td>15632.595996</td>\n",
       "      <td>24602.431497</td>\n",
       "      <td>7952.051858</td>\n",
       "      <td>6769.222448</td>\n",
       "      <td>3235.793456</td>\n",
       "      <td>...</td>\n",
       "      <td>9360.924485</td>\n",
       "      <td>6577.091096</td>\n",
       "      <td>14005.596934</td>\n",
       "      <td>12681.484609</td>\n",
       "      <td>7040.423500</td>\n",
       "      <td>7709.460242</td>\n",
       "      <td>9980.664775</td>\n",
       "      <td>4863.740941</td>\n",
       "      <td>1559</td>\n",
       "      <td>MT-RNR2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000209082</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>75</td>\n",
       "      <td>MT-TL1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 320 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  GSM2100046_291_1_counts  GSM2099989_107_5_counts  \\\n",
       "0  ENSG00000210049                 0.000000                 3.769992   \n",
       "1  ENSG00000211459               926.463645              1034.623715   \n",
       "2  ENSG00000210077                 0.000000                 0.000000   \n",
       "3  ENSG00000210082              5365.827728              7810.573375   \n",
       "4  ENSG00000209082                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100039_255_2_counts  GSM2100240_845_6_counts  GSM2100242_864_4_counts  \\\n",
       "0                 6.247799                 1.786674                 5.713135   \n",
       "1              2263.769598              2415.808088              1596.912571   \n",
       "2                 0.000000                 0.000000                 0.000000   \n",
       "3              8809.561419             12461.947384             15632.595996   \n",
       "4                 0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100193_821_8_counts  GSM2100273_998_4_counts  GSM2099965_024_1_counts  \\\n",
       "0                 5.205140                 2.829007                 2.867638   \n",
       "1              2378.154201              2125.236620               901.057693   \n",
       "2                 0.000000                 0.000000                 0.000000   \n",
       "3             24602.431497              7952.051858              6769.222448   \n",
       "4                 0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100069_352_1_counts  ...  GSM2100032_243_1_counts  \\\n",
       "0                 2.821041  ...                 7.064852   \n",
       "1               781.230165  ...              1831.329630   \n",
       "2                 0.000000  ...                 0.000000   \n",
       "3              3235.793456  ...              9360.924485   \n",
       "4                 0.000000  ...                 0.000000   \n",
       "\n",
       "   GSM2099966_024_4_counts  GSM2100230_839_5_counts  GSM2100170_601_4_counts  \\\n",
       "0                 7.307869                 6.633206                11.450310   \n",
       "1              1452.967437              2684.458441              1672.083082   \n",
       "2                 0.000000                 0.000000                 1.683172   \n",
       "3              6577.091096             14005.596934             12681.484609   \n",
       "4                 0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100107_799_03_counts  GSM2100182_668_1_counts  GSM2099968_054_1_counts  \\\n",
       "0                  1.259363                 1.292854                 2.895095   \n",
       "1               1013.929710               946.021915              1165.762835   \n",
       "2                  0.000000                 0.000000                 0.000000   \n",
       "3               7040.423500              7709.460242              9980.664775   \n",
       "4                  0.000000                 0.000000                 0.000000   \n",
       "\n",
       "   GSM2100023_215_1_counts  transcript_length  Gene_name  \n",
       "0                 1.489433                 71      MT-TF  \n",
       "1              1959.141027                954    MT-RNR1  \n",
       "2                 0.000000                 69      MT-TV  \n",
       "3              4863.740941               1559    MT-RNR2  \n",
       "4                 0.000000                 75     MT-TL1  \n",
       "\n",
       "[5 rows x 320 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for column in ipsc_samples_df.columns:\n",
    "    if column not in ['Gene','transcript_length' ,'Gene_name']:\n",
    "        ipsc_samples_df[column] =  ipsc_samples_df[column]/ scaling_fac_dic[column]\n",
    "\n",
    "ipsc_samples_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(411430, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene1</th>\n",
       "      <th>Gene2</th>\n",
       "      <th>PPI_val</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>411428</th>\n",
       "      <td>DUX4</td>\n",
       "      <td>TBB2A</td>\n",
       "      <td>0.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411429</th>\n",
       "      <td>DICER</td>\n",
       "      <td>ADT2</td>\n",
       "      <td>0.63</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene1  Gene2  PPI_val\n",
       "411428   DUX4  TBB2A     0.56\n",
       "411429  DICER   ADT2     0.63"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hippie = pd.read_csv('./data/hippie_current.txt',sep = '\\t',header=None)\n",
    "hippie.columns = ['Gene1', 'Gene1_Entrezid', 'Gene2', 'Gene2_Entrezid' ,'PPI_val', 'Procedure']\n",
    "hippie = hippie[['Gene1','Gene2','PPI_val']]\n",
    "# hippie = hippie[hippie['PPI_val']>0]\n",
    "hippie = hippie.replace({'_HUMAN':''}, regex=True)\n",
    "print(hippie.shape)\n",
    "hippie_genes = set(hippie['Gene1'].tolist())|set(hippie['Gene2'].tolist())\n",
    "hippie.tail(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HGNC ID</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Approved name</th>\n",
       "      <th>Previous symbols</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>ENSB_id</th>\n",
       "      <th>UniProt ID(supplied by UniProt)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HGNC:5</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>alpha-1-B glycoprotein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000121410</td>\n",
       "      <td>P04217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HGNC:37133</td>\n",
       "      <td>A1BG-AS1</td>\n",
       "      <td>A1BG antisense RNA 1</td>\n",
       "      <td>NCRNA00181, A1BGAS, A1BG-AS</td>\n",
       "      <td>FLJ23569</td>\n",
       "      <td>ENSG00000268895</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      HGNC ID Gene_name           Approved name             Previous symbols  \\\n",
       "0      HGNC:5      A1BG  alpha-1-B glycoprotein                          NaN   \n",
       "1  HGNC:37133  A1BG-AS1    A1BG antisense RNA 1  NCRNA00181, A1BGAS, A1BG-AS   \n",
       "\n",
       "   Synonyms          ENSB_id UniProt ID(supplied by UniProt)  \n",
       "0       NaN  ENSG00000121410                          P04217  \n",
       "1  FLJ23569  ENSG00000268895                             NaN  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#https://www.genenames.org/download/custom/\n",
    "gene_names = pd.read_csv('./iPSC/gene_names.txt',sep='\\t')\n",
    "gene_names = gene_names.rename(columns= {'Ensembl ID(supplied by Ensembl)': 'ENSB_id','Approved symbol':'Gene_name'})\n",
    "gene_names.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "rev_ensg_dict = dict(zip(gene_names.ENSB_id,gene_names.Gene_name))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "411430\n",
      "411430\n"
     ]
    }
   ],
   "source": [
    "ensg_dict = dict(zip(gene_names.Gene_name, gene_names.ENSB_id))\n",
    "\n",
    "gene1_ensg = []\n",
    "gene2_ensg = []\n",
    "ensg_dict_keys = list(ensg_dict.keys())\n",
    "for index,row in hippie.iterrows():\n",
    "    gene1 = row['Gene1']\n",
    "    gene2 = row['Gene2']\n",
    "    if gene1 in (ensg_dict_keys):\n",
    "        gene1_ensg.append(ensg_dict[gene1])\n",
    "    else:\n",
    "        gene1_ensg.append(0)\n",
    "    if gene2 in (ensg_dict_keys):\n",
    "        gene2_ensg.append(ensg_dict[gene2])\n",
    "    else:\n",
    "        gene2_ensg.append(0)\n",
    "        \n",
    "\n",
    "    \n",
    "    \n",
    "\n",
    "print(len(gene1_ensg))\n",
    "print(len(gene2_ensg))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "hippie['Gene1_ENSG'] = gene1_ensg\n",
    "hippie['Gene2_ENSG'] = gene2_ensg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "50173\n",
      "6601\n",
      "6540\n"
     ]
    }
   ],
   "source": [
    "ipsc_genes = ipsc_samples_df['Gene'].tolist()\n",
    "hippie_genes = set(hippie['Gene1_ENSG'].tolist()) or set(hippie['Gene2_ENSG'].tolist())\n",
    "common_genes = set(ipsc_genes) & set(hippie_genes)\n",
    "print(len(ipsc_genes))\n",
    "print(len(hippie_genes))\n",
    "print(len(common_genes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6391\n",
      "6391\n"
     ]
    }
   ],
   "source": [
    "hippie_short = hippie[(hippie['Gene1_ENSG'].isin(common_genes))&(hippie['Gene2_ENSG'].isin(common_genes))]\n",
    "hippie_short_genes = set(hippie_short['Gene1_ENSG'].tolist())|set(hippie_short['Gene2_ENSG'].tolist())\n",
    "print(len(hippie_short_genes))\n",
    "common_genes_short = set(ipsc_genes) & set(hippie_short_genes)\n",
    "common_genes_short = list(common_genes_short)\n",
    "print(len(common_genes_short))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ENSG00000129654',\n",
       " 'ENSG00000198625',\n",
       " 'ENSG00000128585',\n",
       " 'ENSG00000124479',\n",
       " 'ENSG00000148600']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(common_genes_short)[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6391\n"
     ]
    }
   ],
   "source": [
    "rev_ensg_dict\n",
    "common_genes_short_names = []\n",
    "\n",
    "for i in common_genes_short:\n",
    "    common_genes_short_names.append(rev_ensg_dict[i])\n",
    "    \n",
    "print(len(common_genes_short_names))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6391\n"
     ]
    }
   ],
   "source": [
    "ipsc_samples_df_short = ipsc_samples_df[ipsc_samples_df['Gene'].isin(common_genes_short)]\n",
    "print(len(ipsc_samples_df_short['Gene_name'].drop_duplicates().tolist()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# shortlisting only with unique hippie genes from the whole samplespace\n",
    "\n",
    "# gtex_shortlisted = pd.DataFrame()\n",
    "# count = 0\n",
    "# with open('./data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_tpm.tsv') as csvfile:\n",
    "#     reader = csv.DictReader(csvfile, delimiter='\\t')\n",
    "#     for row in reader:\n",
    "#         count+=1\n",
    "#         if count % 1000 == 0:\n",
    "#             print(count)\n",
    "            \n",
    "#         if row['Name'].split('.')[0] in common_genes_short:\n",
    "#             gtex_shortlisted = gtex_shortlisted.append(row, ignore_index=True)\n",
    "\n",
    "\n",
    "            \n",
    "# gtex_shortlisted = gtex_shortlisted.rename(index=str,columns={'Description':'Gene_name','Name':'Ensembl_id'})\n",
    "# gtex_shortlisted.to_csv(\"./iPSC/gtex_shortlisted_ipsc.tsv\",sep=\"\\t\",index = False)\n",
    "\n",
    "gtex_shortlisted = pd.read_csv('iPSC/gtex_shortlisted_ipsc.tsv',sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6380, 11690)"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_shortlisted.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "129 for ensg -\n",
    "128 for genenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-1117F-0426-SM-5EGHI</th>\n",
       "      <th>GTEX-1117F-0526-SM-5EGHJ</th>\n",
       "      <th>GTEX-1117F-0626-SM-5N9CS</th>\n",
       "      <th>GTEX-1117F-0726-SM-5GIEN</th>\n",
       "      <th>GTEX-1117F-1326-SM-5EGHH</th>\n",
       "      <th>GTEX-1117F-2226-SM-5N9CH</th>\n",
       "      <th>GTEX-1117F-2426-SM-5EGGH</th>\n",
       "      <th>GTEX-1117F-2526-SM-5GZY6</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-1226-SM-5N9CK</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2426-SM-5E44I</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "      <th>Ensembl_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NOC2L</td>\n",
       "      <td>90.38</td>\n",
       "      <td>92.83</td>\n",
       "      <td>94.7</td>\n",
       "      <td>67.22</td>\n",
       "      <td>26.55</td>\n",
       "      <td>36.08</td>\n",
       "      <td>51.46</td>\n",
       "      <td>85.61</td>\n",
       "      <td>93.03</td>\n",
       "      <td>...</td>\n",
       "      <td>42.5</td>\n",
       "      <td>60.75</td>\n",
       "      <td>41.34</td>\n",
       "      <td>51.55</td>\n",
       "      <td>55.38</td>\n",
       "      <td>70.85</td>\n",
       "      <td>57.98</td>\n",
       "      <td>74.59</td>\n",
       "      <td>39.16</td>\n",
       "      <td>ENSG00000188976.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 11690 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene_name  GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
       "0     NOC2L                     90.38                     92.83   \n",
       "\n",
       "   GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
       "0                      94.7                     67.22   \n",
       "\n",
       "   GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  \\\n",
       "0                     26.55                     36.08   \n",
       "\n",
       "   GTEX-1117F-2226-SM-5N9CH  GTEX-1117F-2426-SM-5EGGH  \\\n",
       "0                     51.46                     85.61   \n",
       "\n",
       "   GTEX-1117F-2526-SM-5GZY6  ...  GTEX-ZZPU-1226-SM-5N9CK  \\\n",
       "0                     93.03  ...                     42.5   \n",
       "\n",
       "   GTEX-ZZPU-1326-SM-5GZWS  GTEX-ZZPU-1426-SM-5GZZ6  GTEX-ZZPU-1826-SM-5E43L  \\\n",
       "0                    60.75                    41.34                    51.55   \n",
       "\n",
       "   GTEX-ZZPU-2126-SM-5EGIU  GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2426-SM-5E44I  \\\n",
       "0                    55.38                    70.85                    57.98   \n",
       "\n",
       "   GTEX-ZZPU-2626-SM-5E45Y  GTEX-ZZPU-2726-SM-5NQ8O         Ensembl_id  \n",
       "0                    74.59                    39.16  ENSG00000188976.6  \n",
       "\n",
       "[1 rows x 11690 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_shortlisted.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ENSG00000241563'"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'ENSG00000241563.2'.split('.')[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6380\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(6330, 320)"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_genes = gtex_shortlisted['Gene_name'].drop_duplicates().tolist()\n",
    "print(len(gtex_genes))\n",
    "ipsc_samples_df_short = ipsc_samples_df_short[ipsc_samples_df_short['Gene_name'].isin(gtex_genes)]\n",
    "ipsc_samples_df_short.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "essential_gene_df = pd.read_csv('./iPSC/TableS2.txt',sep='\\t',header = None)\n",
    "essential_gene_df = essential_gene_df.rename(columns={0:'Gene_name',1:'HGNC'})\n",
    "essential_genes = set(essential_gene_df['Gene_name'].tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "278"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(gtex_genes)&essential_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "ipsc_samples_df_short = ipsc_samples_df_short.rename(index=str,columns={'Gene':'Ensembl_id'})\n",
    "\n",
    "ipsc_samples_df_short.to_csv(\"./iPSC/ipsc_samples_df_short.csv\",index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ensembl_id</th>\n",
       "      <th>GSM2100046_291_1_counts</th>\n",
       "      <th>GSM2099989_107_5_counts</th>\n",
       "      <th>GSM2100039_255_2_counts</th>\n",
       "      <th>GSM2100240_845_6_counts</th>\n",
       "      <th>GSM2100242_864_4_counts</th>\n",
       "      <th>GSM2100193_821_8_counts</th>\n",
       "      <th>GSM2100273_998_4_counts</th>\n",
       "      <th>GSM2099965_024_1_counts</th>\n",
       "      <th>GSM2100069_352_1_counts</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM2100032_243_1_counts</th>\n",
       "      <th>GSM2099966_024_4_counts</th>\n",
       "      <th>GSM2100230_839_5_counts</th>\n",
       "      <th>GSM2100170_601_4_counts</th>\n",
       "      <th>GSM2100107_799_03_counts</th>\n",
       "      <th>GSM2100182_668_1_counts</th>\n",
       "      <th>GSM2099968_054_1_counts</th>\n",
       "      <th>GSM2100023_215_1_counts</th>\n",
       "      <th>transcript_length</th>\n",
       "      <th>Gene_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>ENSG00000177951</td>\n",
       "      <td>24.341514</td>\n",
       "      <td>24.141653</td>\n",
       "      <td>23.23695</td>\n",
       "      <td>25.840601</td>\n",
       "      <td>20.657215</td>\n",
       "      <td>26.180232</td>\n",
       "      <td>29.30923</td>\n",
       "      <td>26.81183</td>\n",
       "      <td>25.311488</td>\n",
       "      <td>...</td>\n",
       "      <td>25.501669</td>\n",
       "      <td>28.291676</td>\n",
       "      <td>20.414624</td>\n",
       "      <td>22.821525</td>\n",
       "      <td>21.188479</td>\n",
       "      <td>27.16633</td>\n",
       "      <td>23.508748</td>\n",
       "      <td>21.033895</td>\n",
       "      <td>2916</td>\n",
       "      <td>BET1L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 320 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Ensembl_id  GSM2100046_291_1_counts  GSM2099989_107_5_counts  \\\n",
       "43  ENSG00000177951                24.341514                24.141653   \n",
       "\n",
       "    GSM2100039_255_2_counts  GSM2100240_845_6_counts  GSM2100242_864_4_counts  \\\n",
       "43                 23.23695                25.840601                20.657215   \n",
       "\n",
       "    GSM2100193_821_8_counts  GSM2100273_998_4_counts  GSM2099965_024_1_counts  \\\n",
       "43                26.180232                 29.30923                 26.81183   \n",
       "\n",
       "    GSM2100069_352_1_counts  ...  GSM2100032_243_1_counts  \\\n",
       "43                25.311488  ...                25.501669   \n",
       "\n",
       "    GSM2099966_024_4_counts  GSM2100230_839_5_counts  GSM2100170_601_4_counts  \\\n",
       "43                28.291676                20.414624                22.821525   \n",
       "\n",
       "    GSM2100107_799_03_counts  GSM2100182_668_1_counts  \\\n",
       "43                 21.188479                 27.16633   \n",
       "\n",
       "    GSM2099968_054_1_counts  GSM2100023_215_1_counts  transcript_length  \\\n",
       "43                23.508748                21.033895               2916   \n",
       "\n",
       "    Gene_name  \n",
       "43      BET1L  \n",
       "\n",
       "[1 rows x 320 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ipsc_samples_df_short.head(1)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
